src/helm-frontend/src/components/LeaderboardTables.tsx

```diff
@@ -32,10 +34,23 @@ export default function LeaderboardTables({
     ...groupsTables[activeGroup],
   });
   const [sortDirection, setSortDirection] = useState<number>(1);
+  const [filteredModels, setFilteredModels] =
+    useState<string[]>(modelsToFilter);
 
   useEffect(() => {
     setActiveGroupsTable({ ...groupsTables[activeGroup] });
-  }, [activeGroup, groupsTables]);
+    // upon receiving and setting data for current table, use sort to figure out n top models
+    if (nModelsToAutoFilter) {
+      const activeRows = groupsTables[0].rows;
+      const sortedRows = activeRows.sort((a, b) => {
+        return Number(b[1].value) - Number(a[1].value);
+      });
+      // Get the top N rows
```

Comment:
top `numModelsToAutoFilter` (actually, maybe a better name would be `numModelsToShow`)